### ISSUE TITLE
Support for federated learning

### ISSUE THREAD

**Issue by SimonBiggs** [dicomrtai/dicomrtai#16]
Hi @pchlap,

Have you seen the following?

https://github.com/adap/flower

... I haven't put much thought into it (at all), but it doesn't seem too crazy to think that this could be set up in such a way that it is designed to be pluggable into that...

**Comment by pchlap** [comment]
I haven't seen that, thanks for the link. I'll show it to the AusCAT team.

Looks like it uses gRPC as well so it definitely could plug in to this quite nicely...

**Comment by SimonBiggs** [comment]
I really like their directory set up... Having, docker, python, and proto code all under `src` like that... 

https://github.com/adap/flower/tree/main/src

**Comment by SimonBiggs** [comment]
Of particular interest is the following paper:

https://arxiv.org/abs/1905.10497

Which has been implemented here:

https://github.com/adap/flower/blob/5afdc061e76aac446ddf2374a7a97f5ed647ef53/src/py/flwr/server/strategy/aggregate.py#L53-L68

**Comment by pchlap** [comment]
> As I mull on this, I am wondering if the best approach forward is to utilise `flwr` to run the "server/machine-learning client" backend connection. Even though, for now, I only have one machine learning client, if I ever did want to add more machine learning clients to the set-up this may be quite a tidy way to do it. And it would mean that the code for your use case, and my use case, would be more closely aligned.

Agreed... if you think this works well for your use case. In the interest of building the tool that solves one use case first I would be happy to investigate how we might plug flwr in at a later time if/when it might be needed.
